spec/features/users_can_sign_in_spec.rb

```diff
@@ -45,37 +45,62 @@ def log_in_via_form(user, remember_me: false)
     end
 
     context "user has a confirmed mobile number" do
-      scenario "successful sign in via header link" do
-        # Given a user with 2FA enabled and a confirmed mobile number exists
-        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.current)
-
-        # When I log in with that user's email and password
-        visit root_path
-        log_in_via_form(user)
+      context "and the OTP is correct" do
+        scenario "successful sign in via header link" do
+          # Given a user with 2FA enabled and a confirmed mobile number exists
+          user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.current)
+          allow_any_instance_of(User).to receive(:current_otp).and_return(123456)
+          allow_any_instance_of(User).to receive(:validate_and_consume_otp!).and_return(true)
+
+          # When I log in with that user's email and password
+          travel_to Time.new(2023, 11, 29, 10, 0o0, 0o0) do
+            visit root_path
+            log_in_via_form(user)
+
+            # Then there should be no link to check my mobile number
+            expect(page).not_to have_link "Check your mobile number is correct"
+
+            # And I should receive an OTP to my mobile number
+            expect(notify_client).to have_received(:send_sms).with({
+              phone_number: user.mobile_number,
+              template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
+              personalisation: {otp: 123456}
+            })
+          end
+
+          # When I enter the one-time password before it expires
+          travel_to Time.new(2023, 11, 29, 10, 0o5, 29) do
+            fill_in "Please enter your six-digit verification code", with: 123456
+            click_on "Continue"
+          end
+
+          # Then I should be logged in
+          expect(page).to have_content("Signed in successfully.")
+          expect(page).to have_link(t("header.link.sign_out"))
 
-        # Then there should be no link to check my mobile number
-        expect(page).not_to have_link "Check your mobile number is correct"
+          # And I should be at the home page
+          expect(page).to have_content("You can search by RODA, Partner Organisation, BEIS, or IATI identifier, or by the activity's title")
+        end
+      end
 
-        otp_at_time_of_login = user.current_otp
-        # And I should receive an OTP to my mobile number
-        expect(notify_client).to have_received(:send_sms).with({
-          phone_number: user.mobile_number,
-          template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
-          personalisation: {otp: otp_at_time_of_login}
-        })
+      context "and the OTP has expired (Configured as 5 minutes, but appears to be 5 minutes 30 seconds?)" do
```

Comment:
I would remove the question mark - we are already expressing doubt with "appears".

Also not sure I'd put this in the spec description or a comment within the spec, but that may be a matter of preference.